### PR TITLE
Add documentation to limit orderline quantity

### PR DESCRIPTION
**Pull request by bjarnef**
This PR add a how-to documentation for limiting orderline quantity on product add and orderline quantity change based on the product stock value and the current quantities of the product in cart.

New PR from closed https://github.com/vendrhub/vendr-documentation/pull/29

### CODE REVIEW

File: content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+---
+title: Limit Order Line Quantity
+description: How-To Guide to limit order line quantity in Vendr, the eCommerce solution for Umbraco v8+
+---
+
+In this guide we will be looking at Validation events in Vendr to be able to limit order line quantity based on the existing stock value on the product and the existing quantities of the product in the cart.
+
+## ProductAddValidationHandler
+
+When adding a product to cart we need to verify the product is in stock and the customer not already has the remaining quantities in cart.
```
Comment (mattbrailsford) [inline content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md:10]: ..customer **does not already have** the ...

File: content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        _context = context;
+    }
+
+    public override void Validate(ValidateOrderProductAdd evt)
+    {
+        var order = evt.Order;
+        var productReference = evt.ProductReference;
+
+        var product = _context.UmbracoContext.Content.GetById(new Guid(productReference));
+        var stock = product.Value<decimal?>("stock");
```
Comment (mattbrailsford) [inline content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md:28]: should probably not bother using the property value here, I'd recommend using the `IProductService` service instead, which would need injecting into the handler, then call

     _productService.GetProductStock(productReference);

File: content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        if (stock.HasValue && totalQuantities >= stock.Value)
+            evt.Fail($"Only {stock} quantities can be purchased for {productReference}");
+    }
+}
+
+````
+
+## OrderLineQuantityValidationHandler
+
+Furthermore when changing order line quantity on cart page, we need to ensure to quantities being changed to are in stock.
```
Comment (mattbrailsford) [inline content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md:41]: - ...changing **the** order line...
- ...ensure **the** quantities being changed are in stock.

File: content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        if (stock.HasValue && evt.Quantity.To > stock.Value)
+            evt.Fail($"Only {stock} quantities can be purchased for {productReference}");
+    }
+}
+
+````
+
+## Register event handlers
+
+Finally we need to register the Vendr event handlers in our `StoreComposer`.
```
Comment (mattbrailsford) [inline content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md:70]: I'd probably end this is "event handlers in a `IComposer`" as the way this reads suggest it's something they already have in place yet it's not mentioned anywhere else.

File: content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md (hunk 2 above)
Comment (bjarnef) [reply content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md:28]: I haven't tested that, but I guess it should be working in a similar way.

Comment (bjarnef) [reply content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md]: Fixed in https://github.com/vendrhub/vendr-documentation/pull/32/commits/828a28319531d4507abfa77ce18fdba19c8ac808

**Comment by bjarnef** [comment]
@mattbrailsford the suggestions have be resolved now 👍

**Pull request merged**